abstract/imaging_device.py

```diff
@@ -70,21 +142,24 @@ def close():
         # Shutdown / clean up.
         
     def get_properties(self, property_names):
-        """Return a dictionary of {'property_name': value} for the list of properties requested.
+        """Return a dictionary of {'property_name': HWProperty} for the list of properties requested.
```

Comment:
Why return HWProperty here instead of the property value? (is this meant to be used differently from get_property_info?)

Reply:
I thought this function was going to return a list that only contained the names of all the properties, or maybe name and description. If we return all the properties to the user then they'd probably also have to be in the dictionary of properties that the user was interested in, since now they'll have all of them and we won't know which ones they really cared about. For some cameras this could be hundreds of properties, and might make it difficult to stay on top of which properties changed when the user called `set_properties()`.

Reply:
I'm not really following here.. my intent was that this method is used to get the _values_ of specific properties requested by the user, like:
```
cam.get_properties(['exposure_time', 'binning_x', 'binning_y'])
# returns:  {
#    'exposure_time': 0.002,
#    'binning_x': 4,
#    'binning_y': 4,
# }
```
..and since the user has control over which values are being requested, I don't think it should be necessary for us to track `self.properties_used` ?

If the user wants to know what properties are available, then they'd use `get_property_info`.

Reply:
Sorry, my mistake, I crossed `get_properties` and `get_property_info` in my mind.

To answer the question that you originally asked. My thinking was to return `HWProperty` instead of value so that user could easily get all associated information about the property at the same time, for example the description, type, range, etc.. They'd also be allowed to change the `HWProperty` object (as it would be a copy) and pass it back to `set_properties`.